otwcode/otwarchive
# AO3-4718 Tag comments visible to logged-out users with the comment ID.

## Issue

https://otwarchive.atlassian.net/browse/AO3-4718

## Purpose

When a user knows the ID of a comment on a tag, there is a way for the user to view the contents of the comment, even if they aren't a tag wrangler. This pull request prevents that from happening.

## Testing

There are two things to check. The first involves viewing replies to comments on tags:

1. Log in as a tag wrangler.
2. Comment on a tag.
3. Reply to your first comment.
4. Click on the "Thread" button on the reply that you just created.
5. Copy the link to the thread.
6. Log out.
7. Paste the link to the thread and try to access it again.

You should be redirected to the homepage with a message saying you don't have permission. You should not see the comment that you just posted.

The second thing to test is the cancel_comment_edit view:

1. Log in as tag wrangler.
2. Find any comment on a tag and view the comment page.
3. The number at the end of the url is the comment id. Either memorize it, or copy it.
4. Log out.
5. Try to access the page "/comments/cancel_comment_edit.js?id=<comment id>" on whatever server you're testing on.

Again, you should be redirected to the homepage with a message saying you don't have permission. You should not see javascript containing the escaped HTML of the tag comment.
